californium-core/src/main/java/org/eclipse/californium/core/network/UdpMatcher.java

```diff
@@ -387,12 +388,6 @@ public void run() {
 								LOGGER.debug("ignoring notify for pending cancel {}!", response);
 								return;
 							}
-							// we have received the expected response for the
-							// original request
-							idByMID = KeyMID.fromOutboundMessage(currentRequest);
-							if (exchangeStore.remove(idByMID, exchange) != null) {
-								LOGGER.debug("closed open request [{}]", idByMID);
-							}
```

Comment:
That's removed by the exchange.complete anyway. May result only in some more
"ignoring response {}, exchange not longer matching!"